contract/tests.rs

```diff
@@ -0,0 +1,216 @@
+
+    // Assert
+    assert_eq!(contract.max_points_per_sender, max_points_per_sender);
+}
+
+#[ink::test]
+fn is_member_works() {
+    // Arrange
+    let (mut contract, accounts) = init(None);
+    contract.members.push(accounts.alice);
+
+    // Act
+    let result_true = contract.is_member(&accounts.alice);
+    let result_false = contract.is_member(&accounts.bob);
+
+    // Assert
+    assert_eq!(result_true, true);
+    assert_eq!(result_false, false);
+}
+
+#[ink::test]
+fn check_admin() {
+    // Arrange
+    let (contract, accounts) = init(None);
+
+    // Act
+    set_sender(accounts.alice);
+    let result_ok = contract.check_admin();
+
+    set_sender(accounts.bob);
+    let result_fails = contract.check_admin();
+
+    // Assert
+    assert_eq!(result_ok, Ok(()));
+    assert_eq!(result_fails, Err(ContractError::AdminRequired))
+}
+
+#[ink::test]
+fn add_member() {
+    // Arrange
+    let (mut contract, accounts) = init(None);
+    set_sender(accounts.alice);
+
+    // Act
+    let result_ok = contract.add_member(accounts.bob);
+    let result_fails = contract.add_member(accounts.bob);
+
+    // Assert
+    assert_eq!(result_ok, Ok(()));
+    assert_eq!(result_fails, Err(ContractError::MemberAlreadyExists));
+}
+
+#[ink::test]
+fn award_works() {
+    // Arrange
+    let (mut contract, accounts) = init(None);
+    set_sender(accounts.alice);
+    contract.members.push(accounts.bob);
+    contract.members.push(accounts.frank);
+
+    // Act
+    let result_award_bob_ok = contract.award(accounts.bob, 2);
+    let result_award_frank_ok = contract.award(accounts.frank, 3);
+
+    // Assert
+    assert_eq!(result_award_bob_ok, Ok(()));
+    assert_eq!(result_award_frank_ok, Ok(()));
+    assert_eq!(contract.total_points, 5);
+    assert_eq!(contract.recipients.len(), 2);
+    assert_eq!(contract.senders.len(), 1);
+}
+
+#[ink::test]
+fn award_with_null_points_amount_fails_() {
+    // Arrange
+    let (mut contract, accounts) = init(None);
+    set_sender(accounts.alice);
+
+    // Act
+    let result_fails = contract.award(accounts.bob, 0);
+
+    // Assert
+    assert_eq!(
+        result_fails,
+        Err(ContractError::AwardPointsMustBeGreaterThanZero)
+    );
+}
+
+#[ink::test]
+fn award_with_same_sender_and_recipient_fails() {
+    // Arrange
+    let (mut contract, accounts) = init(None);
+    set_sender(accounts.alice);
+
+    // Act
+    let result_fails = contract.award(accounts.alice, 1);
+
+    // Assert
+    assert_eq!(result_fails, Err(ContractError::CannotAwardYourself));
+}
+
+#[ink::test]
+fn award_with_points_amount_exceeding_the_maximum() {
+    // Arrange
+    let (mut contract, accounts) = init(Some(5));
+    set_sender(accounts.alice);
+    contract.members.push(accounts.bob);
+
+    // Act
+    let result_fails = contract.award(accounts.bob, 10);
+
+    // Assert
+    assert_eq!(result_fails, Err(ContractError::MaxPointsPerSenderExceeded));
+}
+
+#[ink::test]
+fn distribute_rewards_works() {
```

Comment:
Here we are missing the asserts where we check that each member got the expected amount of funds

Reply:
We have to do a E2E test to check that

Reply:
:arrow_forward: Pending to do E2E tests